packages/story-editor/src/types/configProvider.ts

```diff
@@ -0,0 +1,306 @@
+      parent?: number;
+      slug?: string;
+    }
+  ) => Promise<Term>;
+  deleteMedia?: (id: number) => Promise<boolean>;
+  deletePageTemplate?: (id: number) => Promise<boolean>;
+  getAuthors?: () => Promise<Author[]>;
+  getCurrentUser?: () => Promise<User>;
+  getCustomPageTemplates?: (page: number | boolean) => Promise<{
+    hasMore: boolean;
+    templates: PageTemplate[];
+  }>;
+  getFonts?: (props: GetFontProps) => Promise<Font[]>;
+  getHotlinkInfo?: (link: string) => Promise<HotlinkInfo>;
+  getLinkMetadata?: (link: string) => Promise<LinkMetaData>;
+  getMedia?: (props: {
+    mediaType: string;
+    searchTerm: string;
+    pagingNum: number;
+  }) => Promise<{
+    data: Resource[];
+    headers: Record<string, string>;
+  }>;
+  getMediaById?: (id: number) => Promise<Resource>;
+  // @todo Not certain if correct.
+  getMediaForCorsCheck?: () => Promise<Resource[]>;
+  getMutedMediaById?: (id: number) => Promise<VideoResource>;
+  getOptimizedMediaById?: (id: number) => Promise<Resource>;
+  getPageTemplates?: () => Promise<PageTemplate[]>;
+  getPosterMediaById?: (id: number) => Promise<Resource>;
+  getProducts?: () => Promise<Product[]>;
+  getProxyUrl?: (src: string) => string;
+  getStoryById?: (id: number) => Promise<Story>;
+  getTaxonomies?: () => Promise<Taxonomy[]>;
+  getTaxonomyTerm?: (props: {
+    search?: string;
+    per_page?: number;
+  }) => Promise<Term>;
+  saveStoryById?: (data: Story) => Promise<Story>;
+  updateCurrentUser?: (data: {
+    mediaOptimization?: boolean;
+    onboarding?: Record<string, boolean>;
+    trackingOptin?: boolean;
+  }) => Promise<User>;
+  updateMedia?: (id: number, data: Partial<Resource>) => Promise<Resource>;
+  updatePageTemplate?: (
+    id: number,
+    data: Partial<PageTemplate>
+  ) => Promise<PageTemplate>;
+  uploadMedia?: (files: string[], props: UploadMediaProps) => Promise<Resource>;
+}
+
+export interface PostLock {
+  interval: number;
+  showLockedDialog: boolean;
+}
+
+export interface ConfigState {
+  /** Interval in seconds. */
+  autoSaveInterval: number | null;
+  /** Interval in seconds. */
+  localAutoSaveInterval: number;
+  autoSaveLink: string;
+  isRTL: boolean;
+  locale?: Locale;
+  allowedMimeTypes: MimeTypes;
+  storyId: number | null;
+  dashboardLink: string;
+  dashboardSettingsLink: string;
+  generalSettingsLink: string;
+  cdnURL: string;
+  /** Max allowed upload in bytes */
+  maxUpload: number;
+  capabilities: Capabilities;
+  metaData: MetaData;
+  canViewDefaultTemplates: boolean;
+  /** If to show the 3rd party media in the library */
+  showMedia3p: boolean;
+  encodeMarkup: boolean;
+  ffmpegCoreUrl: string;
+  apiCallbacks: APICallbacks;
+  styleConstants: {
+    topOffset: number;
+    leftOffset: number;
+  };
+  /** Data for when a user tries to edit a story that's already being edited */
+  postLock?: PostLock;
+  postType?: string;
+  revisionLink?: string;
+  // @todo Is this used at all?
+  isDemo?: boolean;
```

Comment:
Are we actually using this? 🤔

Reply:
Looks like we can completely remove this! We're handling the demo story differently these days.